### ISSUE TITLE
Make settings for dangerous overwrite behaviour safer, or more consistent and obvious

### ISSUE THREAD

**Issue by jmurty** [jschneier/django-storages#960]
The different storage backends in `django-storages` have different default behaviour when storing a file that matches an existing file name. Some backends will overwrite an existing file without warning, others will adjust the file name to avoid overwriting existing files.

When the default behaviour is to **overwrite** the existing file, this is dangerous and is likely to be a nasty surprise to users that might expect the behaviour to match Django's built-in storage classes which avoid overwriting files in this way. Django instead adds a suffix to keep the file names unique (via [get_available_name()](https://docs.djangoproject.com/en/3.1/ref/files/storage/#django.core.files.storage.Storage.get_available_name) etc)

Could these potentially dangerous settings be made safer by doing some or all of?
- Add a note section to the documentation for each backend near the top to highlight the overwrite behaviour, maybe for all backends but definitely when the default is to overwrite.
- Make all storage engines safer by making it the default **not** to overwrite files. This would be a disruptive breaking change, but could perhaps be mitigated by...
- Make users explicitly set the overwrite setting when the default would otherwise be to overwrite files. For example, make it an error for users not to explicitly set `AWS_S3_FILE_OVERWRITE` to either `True` or `False`.

---

Here is a summary of the differing settings across storage backends according to the latest documentation. The potentially dangerous overwrite settings are often included half-way through many other settings, where they are easy to miss.

**Unsafe overwrite defaults**

> `AWS_S3_FILE_OVERWRITE` (optional: default is `True`)
> By default files with the same name will overwrite each other. Set this to `False` to have extra characters appended.

> `GS_FILE_OVERWRITE` (optional: default is `True`)
> By default files with the same name will overwrite each other. Set this to `False` to have extra characters appended.

**Safe overwrite defaults**

> `DROPBOX_WRITE_MODE` (optional, default `'add'`)
> Sets the Dropbox WriteMode strategy. Read more in the official docs.

> `AZURE_OVERWRITE_FILES`
> Overwrite an existing file when it has the same name as the file being uploaded. Otherwise, rename it. Default is `False`

**Unknown overwrite defaults**

Overwrite settings are not mentioned for backends:
- FTP
- SFTP
- Apache Libcloud (probably the same as for corresponding backend)
- Digital Ocean (probably but same as AWS S3)

**Comment by jschneier** [comment]
Yep. Agreed. The next version, 1.11, will print a warning and the default will change in 2.

**Comment by jmurty** [comment]
Excellent! Thanks for those upcoming changes @jschneier
